File: services/order-service/src/main/java/io/minishop/order/reconciliation/ReconciliationProperties.java
package io.minishop.order.reconciliation;

import org.springframework.boot.context.properties.ConfigurationProperties;

@ConfigurationProperties(prefix = "mini-shop.reconciliation")
public record ReconciliationProperties(
        boolean enabled,
        long intervalMs,
        long lookbackMinutes,
        int batchSize
) {}

File: services/order-service/src/main/kotlin/io/minishop/order/reconciliation/ReconciliationProperties.kt
package io.minishop.order.reconciliation

import org.springframework.boot.context.properties.ConfigurationProperties

@JvmRecord
@ConfigurationProperties(prefix = "mini-shop.reconciliation")
data class ReconciliationProperties(
    val enabled: Boolean,
    val intervalMs: Long,
    val lookbackMinutes: Long,
    val batchSize: Int,
)
